core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildContainerRunner.java

```diff
@@ -81,6 +89,24 @@ protected void objcopy(String... args) {
         runCommand(command, null, null);
     }
 
+    @Override
+    public void addShutdownHook(Process process) {
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            if (process.isAlive()) {
+                try {
+                    Process removeProcess = new ProcessBuilder(
+                            List.of(containerRuntime.getExecutableName(), "rm", "-f", containerName))
+                                    .redirectOutput(ProcessBuilder.Redirect.DISCARD)
+                                    .redirectError(ProcessBuilder.Redirect.DISCARD)
+                                    .start();
+                    removeProcess.waitFor(2_000, TimeUnit.SECONDS);
+                } catch (IOException | InterruptedException e) {
```

Comment:
right some DB are very slow to stop, but 2000 seconds is a lot!  I expect people to think it's hung?

What happens if we don't wait at all, or just wait for some ~10 seconds ?

Also might be nice to include `containerName` in the logged message, and upgrade it from debug to INFO or WARN ?

Reply:
Oh shoot... I mean to have 2 seconds :).

Basically this is just meant to do `docker rm -f build-native` so it's really esoteric, not much the user can do anyway and it doesn't have much of an effect even if it fails.